zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/usermanagement/UserController.java

```diff
@@ -0,0 +1,146 @@
+import io.camunda.zeebe.gateway.protocol.rest.CamundaUserResponse;
+import io.camunda.zeebe.gateway.protocol.rest.CamundaUserWithPasswordRequest;
+import io.camunda.zeebe.gateway.protocol.rest.SearchQueryRequest;
+import io.camunda.zeebe.gateway.protocol.rest.UserSearchResponse;
+import io.camunda.zeebe.gateway.rest.controller.ZeebeRestController;
+import java.util.List;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ProblemDetail;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@ZeebeRestController
+@RequestMapping("/v2/users")
+public class UserController {
+  private final UserService userService;
+
+  public UserController(final UserService userService) {
+    this.userService = userService;
+  }
+
+  @PostMapping(
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE},
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<Object> createUser(
+      @RequestBody final CamundaUserWithPasswordRequest userWithPasswordDto) {
+    try {
+      final CamundaUserResponse camundaUserResponse =
+          mapToCamundaUserResponse(
+              userService.createUserFailIfExists(mapToUserWithPassword(userWithPasswordDto)));
+      return new ResponseEntity<>(camundaUserResponse, HttpStatus.CREATED);
+    } catch (final Exception e) {
+      return handleException(e);
+    }
+  }
+
+  @DeleteMapping(path = "/{id}")
+  public ResponseEntity<Object> deleteUser(@PathVariable final Long id) {
+    try {
+      userService.deleteUser(id);
+      return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    } catch (final Exception e) {
+      return handleException(e);
+    }
+  }
+
+  @GetMapping(
+      path = "/{id}",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE})
+  public ResponseEntity<Object> findUserById(@PathVariable final Long id) {
+    try {
+      final CamundaUserResponse camundaUserResponse =
+          mapToCamundaUserResponse(userService.findUserById(id));
+      return new ResponseEntity<>(camundaUserResponse, HttpStatus.OK);
+    } catch (final Exception e) {
+      return handleException(e);
+    }
+  }
+
+  @PostMapping(
+      path = "/search",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE},
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<Object> findAllUsers(
+      @RequestBody final SearchQueryRequest searchQueryRequest) {
```

Comment:
🔧 This marks the request body as required by default although we don't use it. We can probably mark it as not required?

Reply:
Very good point, thank you. Fixed it :)